File: ccc/2018/j3.java
//https://dmoj.ca/problem/ccc18j3
//CCC '18 J3 - Are we there yet?

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
		List<Integer> cities = new ArrayList<Integer>(5);
		cities.add(0);
		int d1 = s.nextInt();
		cities.add(d1);
		int d2 = s.nextInt();
		cities.add(d1+d2);
		int d3 = s.nextInt();
		cities.add(d1+d2+d3);
		int d4 = s.nextInt();
		cities.add(d1+d2+d3+d4);
		for (int i=0; i<5; i++) {
			for (int city:cities) {
				int out = Math.abs(cities.get(i)-city);
				System.out.print(out+" ");
			}
			System.out.println("");
		}
    	
		s.close();
    }
}

File: ccc/2017/j3.py
#https://dmoj.ca/problem/ccc17j3
#CCC '17 J3 - Exactly Electrical

a,b=list(map(int,input().split()))
c,d=list(map(int,input().split()))
t=int(input())

sum = abs(a-c)+abs(b-d)
if sum<=t:
    if sum%2:
        if t%2:
            print("Y")
        else:
            print("N")
    else:
        if not t%2:
            print("Y")
        else:
            print("N")
else:
    print("N")
